### PR TITLE
cli: add quickstart flow for new users

### PR DESCRIPTION
**Pull request by marcosnils**
This commit changes the CLI so it prints a custom Dagger Cloud trace URL
when the user is not logged in or no `DAGGER_CLOUD_TOKEN` has been set.

Additionally, a custom message gets printed at the end of `dagger init`
and fro any commands that the user supplied the `-v` or `--debug` flags.

Fixes [DEV-4269](https://linear.app/dagger/issue/DEV-4269) [DEV-4268](https://linear.app/dagger/issue/DEV-4268)

Here's some screenshots about how the feature looks:

Log-in message when `init` is called and CLI is not authenticated
image

Full logged out trace URL: 
image

Log-in message when `-v` is supplied: 
image

The custom `/traces/$id` URL will take the user to the quickstart flow which will create a Dagger Cloud organization after logging in.

Signed-off-by: Marcos Lilljedahl <marcosnils@gmail.com>

### CODE REVIEW

File: engine/telemetry/cloud.go
Comment (jedevc) [inline engine/telemetry/cloud.go:23]: lol :laughing:

File: engine/client/client.go
Comment (jedevc) [inline engine/client/client.go:328]: This is really weird from a usability perspective - my initial read of this PR was that we were uploading *all* traces to dagger cloud, and then allowing users to create an organization with them in here. For this case, we shouldn't be printing a URL that looks *anything* like a traces view URL, it definitely confused me - it's not a real trace URL, so we shouldn't pretend it is.

Comment (jedevc) [reply engine/client/client.go:328]: Also note, it's slightly weird that we show the "logged out" message, at the end of a trace in `tui` mode if you're not logged in, but that you don't get the full trace URL if you are logged in?

We should balance that.

Comment (marcosnils) [reply engine/client/client.go:328]: > it's not a real trace URL, so we shouldn't pretend it is.

 cc @shykes

Comment (shykes) [reply engine/client/client.go:328]: As discussed in the issue - the logged in experience did show the Trace URL when we started working on this. But it has since been removed.

Comment (marcosnils) [reply engine/client/client.go]: resolving this as I've removed this code since it requires further discussion

**Comment by marcosnils** [comment]
@jedevc just reverted the `call` prints and changed the URL to be more clear about signup. Mind taking one more quick :eyes: ?

**Review by jedevc — APPROVED** [review]

**Comment by jedevc** [comment]
LGTM once lint/tests are passing :tada:

**Comment by marcosnils** [comment]
mergint as-is since the test failure is a flake.

File: cmd/dagger/call.go
Comment (marcosnils) [reply cmd/dagger/call.go:193]: this was my bad since I didn't rebase with main before merging and this is something helder fixed recently. Will revert in a follow-up PR. Thx for bringing this up alex.

File: cmd/dagger/main.go
Comment (jedevc) [inline cmd/dagger/main.go:184]: Just noticed this now - this should be `!ok && (...)` - since `&&` has a higher precedence, it means that even when logged in, the message is still shown for `debug` and `moduleInitCmd`.

Comment (marcosnils) [reply cmd/dagger/main.go:184]: noted, thx for reporting this one Justin <3

**Pull request merged**
